python/postprocessing/framework/output.py

```diff
@@ -7,48 +7,58 @@
 _rootBranchType2PythonArray = { 'b':'B', 'B':'b', 'i':'I', 'I':'i', 'F':'f', 'D':'d', 'l':'L', 'L':'l', 'O':'B' }
 
 class OutputBranch:
-    def __init__(self, tree, name, rootBranchType, n=1, lenVar=None, title=None, limitedPrecision=False):
+    def __init__(self, tree, name, rootBranchType, n=1, lenVar=None, title=None, limitedPrecision=None, saveBranch=True):
         n = int(n)
         self.buff   = array(_rootBranchType2PythonArray[rootBranchType], n*[0. if rootBranchType in 'FD' else 0])
         self.lenVar = lenVar
         self.n = n
-        self.precision = ROOT.ReduceMantissaToNbitsRounding(limitedPrecision) if limitedPrecision and rootBranchType=='F' else lambda x : x
+        self.precision = ROOT.ReduceMantissaToNbitsRounding(int(limitedPrecision)) if limitedPrecision and rootBranchType=='F' else lambda x : x
         #check if a branch was already there 
-        existingBranch = tree.GetBranch(name)
-        if (existingBranch):
-          self.branch = existingBranch
-          self.branch.SetAddress(self.buff)
-        else:  
-          if lenVar != None:
-            self.branch = tree.Branch(name, self.buff, "%s[%s]/%s" % (name,lenVar,rootBranchType))
-          elif n == 1:
-            self.branch = tree.Branch(name, self.buff, name+"/"+rootBranchType)
-          else:
-            self.branch = tree.Branch(name, self.buff, "%s[%d]/%s" % (name,n,rootBranchType))
-        if title: self.branch.SetTitle(title)
+        if saveBranch:
```

Comment:
Is this option really needed? So we ever needed to produce tree without any branches?

Reply:
Sorry, what option are we talking about?  "saveBranch" is fundamental to how I implemented the change.

Reply:
Yeah, I realized that after I read the code below.